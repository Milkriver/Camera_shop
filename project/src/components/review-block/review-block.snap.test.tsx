import {render} from '@testing-library/react';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import HistoryRouter from '../../components/history-router/history-router';
import ReviewBlock from './review-block';


describe('Component: ReviewBlock', () => {
  test('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      OFFERS: {
        offer: {},
      },
      REVIEWS: {
        reviews: {},
      }
    });

    const history = createMemoryHistory();

    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <ReviewBlock />
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
