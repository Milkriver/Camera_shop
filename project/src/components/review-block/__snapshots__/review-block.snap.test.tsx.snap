// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewBlock should render correctly 1`] = `
<div>
  <section
    class="review-block"
  >
    <div
      class="container"
    >
      <div
        class="page-content__headed"
      >
        <h2
          class="title title--h3"
        >
          Отзывы
        </h2>
        <button
          class="btn"
          type="button"
        >
          Оставить свой отзыв
        </button>
      </div>
    </div>
  </section>
</div>
`;
